deepmd/jax/train/trainer.py

```
@@ -0,0 +1,459 @@
+                label_dict,
+                extended_coord,
+                extended_atype,
+                nlist,
+                mapping,
+                fp,
+                ap,
+            )
+            optimizer.update(grads)
+
+        start_time = time.time()
+        disp_file_fp = open(self.disp_file, "w")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)